gestalt-module/src/main/java/org/terasology/gestalt/module/dependencyresolution/DependencyInfo.java

```diff
@@ -109,25 +108,12 @@ public void setOptional(boolean optional) {
     }
 
     /**
-     * @return The range of supported versions
+     * Returns a predicate that yields true when applied to version that is within the version range described by this dependency information
+     *
+     * @return
      */
-    public VersionRange versionRange() {
-        return new VersionRange(getMinVersion(), getMaxVersion());
-    }
-
     public Predicate<Version> versionPredicate() {
```

Comment:
Thinking about types here.

`Predicate<Version>` is sufficient for gestalt ResolutionAttempt to do its thing. But there are a few other requirements we should consider:

* for troubleshooting purposes, this should at _least_ have a `toString` value that makes more sense than `lambda@123` or whatever. (#15 might also appreciate a rich type, but will also probably find a string sufficient.)
* for gradle compatibility, it needs _either_ this minVersion, maxVersion pair **or** a maven-compatible range descriptor (which is what VersionRange.toString currently provides)

which makes me think we want to keep the return type as VersionRange after all.

Reply:
Let me bring up my question about sum types again... 🙄  nargh

Reply:
I don't think this one needs sum types. One class can implement both `Predicate<Version>` and `ConvertableToMavenRange` or whatever.